examples/testSN.py

```diff
@@ -94,7 +94,62 @@ def test_SNstatenotEmpty(self):
         myDict = self.SN_extincted.SNstate
         for key in myDict.keys():
             assert myDict[key] is not None
-                
+
+
+    def test_attributeDefaults(self):
+        """
+        Check the defaults and the setter properties for rectifySED and
+        modelOutSideRange
+        """
+        snobj = SNObject(ra=30., dec=-60., source='salt2')
+        assert snobj.rectifySED == True
+        assert snobj.modelOutSideTemporalRange == 'zero'
+
+        snobj.rectifySED = False
+        assert snobj.rectifySED == False
+        assert snobj.modelOutSideTemporalRange == 'zero'
+
+    def test_raisingerror_forunimplementedmodelOutSideRange(self):
+        """
+        check that correct error is raised if the user tries to assign an
+        un-implemented model value to
+        `sims.catUtils.supernovae.SNObject.modelOutSideTemporalRange`
+        """
+        snobj = SNObject(ra=30., dec=-60., source='salt2')
+        assert snobj.modelOutSideTemporalRange == 'zero'
+        with self.assertRaises(ValueError) as context:
+            snobj.modelOutSideTemporalRange = 'False'
+        self.assertEqual('Model not implemented, defaulting to zero method\n',
+                         context.exception.message)
+
+    def test_rectifiedSED(self):
+        """
+        Check for an extreme case that the SN seds are being rectified. This is
+        done by setting up an extreme case where there will be negative seds, and
+        checking that this is indeed the case, and checking that they are not
+        negative if rectified.
+        """
+        
+        snobj = SNObject(ra=30., dec=-60., source='salt2')
+        snobj.set(z=0.96, t0=self.mjdobs, x1=-3., x0=1.8e-6)
+        snobj.rectifySED  = False
+        times = np.arange(self.mjdobs - 50., self.mjdobs + 150., 1.)
+        badTimes = []
+        for time in times:
+            sed = snobj.SNObjectSED(time=time,
+                                    bandpass=self.lsstBandPass['r'])
+            if any(sed.flambda < 0.):
+                badTimes.append(time)
+        # Check that there are negative SEDs
+        assert(len(badTimes) > 0)
+        snobj.rectifySED = True
+        for time in badTimes:
+            sed = snobj.SNObjectSED(time=time,
+                                    bandpass=self.lsstBandPass['r'])
+            assert not sed.calcADU(bandpass=self.lsstBandPass['r'],
```

Comment:
The project policy is to use the assert methods from ```unittest```, since they give more explicit error messages when they fail.  In this case you could use

```
self.assertGreater(sed.calcADU(...), 0.0)
```

which will test that the result of ```calcADU()``` is greater than 0